### PR TITLE
fix: [PLG-1820]: Fixed issue with module free plan redirection

### PR DESCRIPTION
**Pull request by devinder-harness**
### Summary

This PR adds new behaviour to `Start Free Plan` buttons for all the modules to `redirect` users to their `Get Strarted` pages with `default` projects selected, if FF `CREATE_DEFAULT_PROJECT` is `on`

#### Screenshots
NA

#### PR Checklist
 
https://harness.atlassian.net/browse/PLG-1820

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[PLG-1820]: https://harness.atlassian.net/browse/PLG-1820?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kapilkumawat86 — APPROVED** [review]

**Review by hekimoglu010 — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

File: src/modules/20-rbac/components/TrialHomePageTemplate/StartTrialTemplate.tsx [hunk 1]
```diff
@@ -77,11 +82,17 @@ const StartTrialComponent: React.FC<StartTrialProps> = startTrialProps => {
...
-      })
+      if (isDefaultProjectCreated) {
+        const moduleUrlWithDefaultProject = getModuleToDefaultURLMap(accountId, module as Module)[module]
+        history.push(
+          moduleUrlWithDefaultProject ? (moduleUrlWithDefaultProject as string) : routes.toHome({ accountId })
+        )
+      } else {
+        history.push({
+          pathname: routes.toModuleHome({ accountId, module }),
+          search: `?modal=${modal}&&experience=${experience}`
```
Comment (knagurski) [inline src/modules/20-rbac/components/TrialHomePageTemplate/StartTrialTemplate.tsx:93]:
```suggestion
          search: `?modal=${modal}&experience=${experience}`
```
```suggestion
          search: `?modal=${modal}&&experience=${experience}`
```

**Review by ishantmehtaharness — APPROVED** [review]

**Review by Sagar-Rajput7 — DISMISSED** [review]

**Review by kapilkumawat86 — DISMISSED** [review]

**Comment by devinder-harness** [comment]
retrigger jest

**Comment by devinder-harness** [comment]
retrigger cypress

**Comment by devinder-harness** [comment]
retrigger coverage

**Comment by devinder-harness** [comment]
retrigger cypress-rest

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Comment by devinder-harness** [comment]
retrigger jest

**Review by Sagar-Rajput7 — APPROVED** [review]

**Comment by devinder-harness** [comment]
retrigger jest

**Pull request merged**